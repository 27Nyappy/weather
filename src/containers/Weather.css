.Error {
  @media only screen and (max-width: 430px) {
    padding: 20px;
    width: 80%;
  }

  @media screen and (min-width: 431px) and (max-width: 600px) {
    padding: 15px 30px;
    width: 80%;
  }

  @media screen and (min-width: 601px) and (max-width: 768px) {
    padding: 15px 30px;
    width: 80%;
  }

  @media screen and (min-width: 769px) and (max-width: 834px) {
    padding: 15px 50px 15px 30px;
    width: 75%;
  }

  @media screen and (min-width: 835px) {
    padding: 15px 50px 15px 30px;
    width: 70%;
  }

  margin: 20px auto;
  line-height: 2rem;
  text-align: center;
  max-width: 600px;
  font-size: 1.08em;
  background-color: #76dbd1;
  box-shadow: 0 0 3px #203f3c95;
  color: #555151;
}