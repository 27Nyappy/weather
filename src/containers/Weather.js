import React, { useState, useEffect } from 'react';
import WeatherForm from '../components/WeatherForm';
import getWeather from '../services/WeatherApi';

const Weather = () => {
  const [zip, updateZip] = useState('');
  const [days, updateDays] = useState('');
  const [weatherInfo, updateWeatherInfo] = useState([]);

  const handleSubmit = event => {
    event.preventDefault();
    getWeather(days, zip)
      .then(res => updateWeatherInfo(res));
  };

  const handleDayChange = ({ target }) => updateDays(target.value);

  const handleZipChange = ({ target }) => updateZip(target.value);

  return (
    <WeatherForm handleSubmit={handleSubmit} handleDayChange={handleDayChange} handleZipChange={handleZipChange} />
  );
};

export default Weather;