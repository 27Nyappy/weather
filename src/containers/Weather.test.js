import React from 'react';
import { shallow } from 'enzyme';
import Weather from './Weather';

describe('Weather Component', () => {
  it('Renders Weather Component', () => {
    const wrapper = shallow(<Weather />);
    expect(wrapper).toMatchSnapshot();
  });
});