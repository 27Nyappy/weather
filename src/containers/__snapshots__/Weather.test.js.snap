// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Component Renders Weather Component 1`] = `
<Fragment>
  <WeatherForm
    handleDayChange={[Function]}
    handleSubmit={[Function]}
    handleZipChange={[Function]}
  />
  <WeatherList
    weatherItems={Array []}
  />
</Fragment>
`;
